adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -200,12 +163,23 @@ public PaymentsRequest createPaymentsRequest(final String merchantAccount,
             paymentsRequest.setShopperReference(customerModel.getCustomerID());
             paymentsRequest.setShopperEmail(customerModel.getContactEmail());
 
-            if(PAYMENT_METHOD_CC.equals(cartData.getAdyenPaymentMethod()) && cartData.getAdyenRememberTheseDetails()) {
-                if(Recurring.ContractEnum.ONECLICK_RECURRING.equals(contractEnum)) {
+            if (PAYMENT_METHOD_CC.equals(cartData.getAdyenPaymentMethod())) {
+                // Set oneclick and recurring preferences
+                paymentsRequest.setEnableRecurring(false);
+                paymentsRequest.setEnableOneClick(false);
+
+                if (Recurring.ContractEnum.ONECLICK_RECURRING.equals(contractEnum)) {
                     paymentsRequest.setEnableRecurring(true);
                     paymentsRequest.setEnableOneClick(true);
                 } else if (Recurring.ContractEnum.ONECLICK.equals(contractEnum)) {
                     paymentsRequest.setEnableOneClick(true);
+                } else if (Recurring.ContractEnum.RECURRING.equals(contractEnum)) {
+                    paymentsRequest.setEnableRecurring(true);
+                }
+
+                // Set storeDetails parameter when shopper selected to have his card details stored
+                if( cartData.getAdyenRememberTheseDetails()) {
+                    paymentsRequest.getPaymentMethod().put(STORE_DETAILS, "true");
```

Comment:
We can use boolean true instead of string true.

Reply:
ok intresting map<string,string> is not very nice but understand why we do it. We could make it map<string,object> so we can support other types. Not sure how the GSON translator will accept this though. we can keep it like this for now.